![trainers in a store](trainers.jpg)

The sports clothing and athleisure industry is a rapidly growing market, valued at around [$193 billion in 2021](https://www.statista.com/statistics/254489/total-revenue-of-the-global-sports-apparel-market/) in 2021 and expected to continue expanding in the coming years. In this analysis, you take on the role of a product analyst for an online sports clothing company, focusing on how to improve revenue. 

You will explore product data such as pricing, reviews, descriptions, ratings, and sales figures. Through comparisons of Adidas and Nike pricing, price segmentation, and the correlation between product descriptions and revenue, you will uncover key insights that can inform marketing and sales strategies.

Additionally, you will analyze the relationship between reviews and revenue, emphasizing the impact of product details and customer feedback on sales. Based on these analyses, you will develop actionable recommendations to help boost the company's performance.

You've been provided with four datasets to investigate:

#  brands.csv

| Columns | Description |
|---------|-------------|
| `product_id` | Unique product identifier |
| `brand` | Brand of the product | 

# finance.csv

| Columns | Description |
|---------|-------------|
| `product_id` | Unique product identifier |
| `listing_price` | Original price of the product | 
| `sale_price` | Discounted price of the product |
| `discount` | Discount off the listing price, as a decimal | 
| `revenue` | Revenue generated by the product |

# info.csv

| Columns | Description |
|---------|-------------|
| `product_name` | Name of the product | 
| `product_id` | Unique product identifier |
| `description` | Description of the product |

# reviews.csv

| Columns | Description |
|---------|-------------|
| `product_id` | Unique product identifier |
| `rating` | Average product rating | 
| `reviews` | Number of reviews for the product |

## Data Overview

In this section, we performed an initial review of the data from the brands, finance, info, and reviews tables. We examined the first few columns of each table to understand the data types and relationships between them. Key connections, such as product-to-brand and revenue-to-reviews, were identified. Additionally, we observed patterns of missing values across columns, highlighting areas that might need data cleaning or imputation before proceeding with deeper analysis.

In [5]:
Select *
From 'info.csv'
limit 5;

column0  ...                                            column2
0                                      product_name  ...                                        description
1                                              None  ...                                               None
2              Women's adidas Originals Sleek Shoes  ...  A modern take on adidas sport heritage, tailor...
3                 Women's adidas Swim Puka Slippers  ...  These adidas Puka slippers for women's come wi...
4  Women's adidas Sport Inspired Questar Ride Shoes  ...  Inspired by modern tech runners, these women's...

[5 rows x 3 columns]

In [6]:
Select *
From 'brands.csv'
limit 5;

column0 column1
0  product_id   brand
1      AH2430    None
2      G27341  Adidas
3      CM0081  Adidas
4      B44832  Adidas

In [7]:
Select *
From 'finance.csv'
limit 5;

product_id  listing_price  sale_price  discount  revenue
0     AH2430            NaN         NaN       NaN      NaN
1     G27341          75.99       37.99       0.5  1641.17
2     CM0081           9.99        5.99       0.4   398.93
3     B44832          69.99       34.99       0.5  2204.37
4     D98205          79.99       39.99       0.5  5182.70

In [8]:
Select *
From 'reviews.csv'
limit 5;

product_id  rating  reviews
0     AH2430     NaN      NaN
1     G27341     3.3     24.0
2     CM0081     2.6     37.0
3     B44832     4.1     35.0
4     D98205     3.5     72.0

In [10]:
SELECT count(*) AS total_rows,
count(info.column2) AS count_description,
count(finance.listing_price) AS count_listing_price,
count(brands.column1) AS count_brand,
count(reviews.rating) AS count_rating

FROM 'info.csv' AS info 
JOIN 'finance.csv' AS finance
ON info.column1 = finance.product_id
JOIN 'brands.csv' AS brands 
ON finance.product_id = brands.column0
JOIN 'reviews.csv' AS reviews
ON finance.product_id = reviews.product_id;

total_rows  count_description  ...  count_brand  count_rating
0        3179               3117  ...         3120          3120

[1 rows x 5 columns]

The analysis revealed that the column headers in the info and brands tables are not clearly identified, preventing direct relationships based on the dataset's column names. Regarding missing values, the info table had the most null entries, with 3117 out of 3179 rows containing usable data. The other tables (finance, reviews, and brands) each had approximately 60 null cells, accounting for only about 2% of the total data, which is relatively minimal and unlikely to significantly impact the analysis.

## 2. Price Comparison Between Adidas and Nike

The dataset contains information exclusively for Adidas and Nike, allowing for a focused comparison of their products. We examined the listing price of each product according to the brand and analyzed the number of shoes available based on the listed price. This comparison highlights the pricing structure of both brands and provides insights into product availability at various price points.

In [13]:
SELECT 
	brands.column1,
	CEIL(listing_price) AS listing_price,
	count(finance.*)
FROM 'brands.csv' AS brands
JOIN 'finance.csv' AS finance
ON brands.column0 = finance.product_id
WHERE finance.listing_price > 0
	AND brands.column1 = 'Adidas'
GROUP BY brands.column1, listing_price
ORDER BY listing_price DESC;

column1  listing_price  count_star()
0   Adidas          300.0             2
1   Adidas          280.0             4
2   Adidas          240.0             5
3   Adidas          230.0             8
4   Adidas          220.0            11
5   Adidas          200.0             8
6   Adidas          190.0             7
7   Adidas          180.0            34
8   Adidas          170.0            27
9   Adidas          160.0            28
10  Adidas          150.0            41
11  Adidas          140.0            36
12  Adidas          130.0            96
13  Adidas          120.0           115
14  Adidas          110.0            91
15  Adidas          100.0            72
16  Adidas           96.0             2
17  Adidas           90.0            89
18  Adidas           86.0             7
19  Adidas           85.0             1
20  Adidas           80.0           322
21  Adidas           76.0           149
22  Adidas           75.0             1
23  Adidas           70.0            87
24  Adidas           66.0           102
25  Adidas           63.0             1
26  Adidas           60.0           211
27  Adidas           56.0           174
28  Adidas           55.0             2
29  Adidas           53.0            43
30  Adidas           50.0           183
31  Adidas           48.0            42
32  Adidas           46.0           163
33  Adidas           45.0             1
34  Adidas           43.0            51
35  Adidas           40.0            81
36  Adidas           38.0            24
37  Adidas           36.0            25
38  Adidas           33.0            24
39  Adidas           30.0            37
40  Adidas           28.0            38
41  Adidas           27.0            18
42  Adidas           25.0            28
43  Adidas           23.0             1
44  Adidas           20.0             8
45  Adidas           18.0             4
46  Adidas           16.0             4
47  Adidas           15.0            27
48  Adidas           13.0            27
49  Adidas           12.0             1
50  Adidas           10.0            11
51  Adidas            9.0             1

In [14]:
SELECT 
	brands.column1,
	CEIL(listing_price) AS listing_price,
	count(finance.*)
FROM 'brands.csv' AS brands
JOIN 'finance.csv' AS finance
ON brands.column0 = finance.product_id
WHERE finance.listing_price > 0
	AND brands.column1 = 'Nike'
GROUP BY brands.column1, listing_price
ORDER BY listing_price DESC;

column1  listing_price  count_star()
0     Nike          200.0             1
1     Nike          190.0             2
2     Nike          180.0             4
3     Nike          170.0            14
4     Nike          160.0            31
5     Nike          150.0             6
6     Nike          140.0            12
7     Nike          130.0            12
8     Nike          120.0            16
9     Nike          110.0            17
10    Nike          100.0            14
11    Nike           95.0             1
12    Nike           90.0            13
13    Nike           85.0             5
14    Nike           80.0            16
15    Nike           79.0             1
16    Nike           75.0             7
17    Nike           70.0             4
18    Nike           65.0             1
19    Nike           60.0             2
20    Nike           50.0             5
21    Nike           48.0             1
22    Nike           45.0             3
23    Nike           40.0             1
24    Nike           30.0             2

The analysis revealed that most of the data corresponds to Adidas sneakers, with the brand offering 52 different prices. In contrast, Nike only has 25 distinct prices listed. Additionally, the count column shows a significant difference in the number of units sold, with Adidas having a larger number of products available for sale across various price points. This highlights the broader pricing range and higher availability of Adidas sneakers compared to Nike.

## 3. Price Labeling Based on Price Segments

In this section, we analyzed the entire dataset and identified price quartiles to create a segmentation based on price ranges. The goal was to categorize the products into four distinct segments: Budget, Average, Expensive, and Elite, with Elite representing the highest-priced items. Products with a price below $46 were labeled as Budget, those between $46 and $60 as Average, between $60 and $90 as Expensive, and above $90 as Elite. Additionally, we examined discount data to determine which brand and segment had the highest number of discounts applied.

In [17]:
CREATE TEMP TABLE price_quartiles AS
    SELECT 
        
        PERCENTILE_CONT(0.25) WITHIN GROUP (ORDER BY finance.listing_price) AS q1,
        PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY finance.listing_price) AS q2,
        PERCENTILE_CONT(0.75) WITHIN GROUP (ORDER BY finance.listing_price) AS q3
    FROM
        'finance.csv' AS finance;
 
SELECT * FROM price_quartiles;

q1     q2     q3
0  45.99  59.99  89.99

In [21]:

SELECT brand.column1,
        count(finance.*)AS 'Count',
        CEIL(SUM(finance.revenue)) AS total_revenue,
        CASE
            WHEN finance.listing_price <= 46 THEN 'Budget'
            WHEN finance.listing_price > 46 AND finance.listing_price <= 60 THEN 'Average'
            WHEN finance.listing_price > 60 AND finance.listing_price <= 90 THEN 'Expensive'
            ELSE 'Elite'
        END AS price_label
FROM 'brands.csv' as brand
JOIN 'finance.csv' AS finance
ON brand.column0 = finance.product_id
WHERE brand.column0 IS NOT NULL AND brand.column1 IS NOT NULL
GROUP BY brand.column1, price_label
ORDER BY total_revenue DESC;

column1  Count  total_revenue price_label
0  Adidas    587      4873735.0       Elite
1  Adidas    759      3507763.0   Expensive
2  Adidas    655      1988124.0     Average
3  Adidas    574      1156999.0      Budget
4    Nike    360       595584.0      Budget
5    Nike    130       177769.0       Elite
6    Nike     47        23527.0   Expensive
7    Nike      8         5405.0     Average

In [22]:
SELECT brand.column1,
        CASE
            WHEN finance.listing_price <= 46 THEN 'Budget'
            WHEN finance.listing_price > 46 AND finance.listing_price <= 60 THEN 'Average'
            WHEN finance.listing_price > 60 AND finance.listing_price <= 90 THEN 'Expensive'
            ELSE 'Elite'
        END AS price_label,
		AVG(finance.discount)*100 AS Average_Discount
FROM 'brands.csv' as brand
JOIN 'finance.csv' AS finance
ON brand.column0 = finance.product_id
WHERE brand.column0 IS NOT NULL AND brand.column1 IS NOT NULL
GROUP BY brand.column1, price_label
ORDER BY Average_Discount DESC;

column1 price_label  Average_Discount
0  Adidas     Average         36.870229
1  Adidas   Expensive         33.412385
2  Adidas      Budget         32.944251
3  Adidas       Elite         30.187394
4    Nike      Budget          0.000000
5    Nike       Elite          0.000000
6    Nike     Average          0.000000
7    Nike   Expensive          0.000000

The analysis categorized the products into four price segments as follows:

- Adidas: The majority of Adidas products fall within the Elite and Expensive categories, with 587 products in the Elite segment and 759 in the Expensive segment. The Budget category had 574 Adidas products, while 655 were in the Average price range.
- Nike: Nike’s distribution is more limited, with most products in the Budget segment (360 units) and fewer in the higher price categories (130 in Elite, 47 in Expensive, and 8 in Average).


Regarding discounts, Adidas showed a higher discount rate across all segments, with the Average segment having the highest average discount at 36.87%. Nike, on the other hand, did not apply any discounts to its products in any segment.

## 4. Correlation Between Revenue and Reviews

In this section, we explored the relationship between revenue and reviews, analyzing both the correlation between revenue and ratings, as well as the number of reviews per product sold. This allowed us to assess how customer feedback (in the form of reviews and ratings) might correlate with the revenue generated by each product. Additionally, we examined the total number of reviews relative to the quantity of shoes sold, providing further insight into customer engagement and satisfaction levels for different products.

In [1]:
SELECT CORR(finance.revenue, reviews.reviews) AS Review_revenue_corr, 
		CORR(finance.revenue, reviews.rating) AS Rate_revenue_corr
FROM 'finance.csv' AS finance
JOIN 'reviews.csv' AS reviews
ON finance.product_id = reviews.product_id

Review_revenue_corr  Rate_revenue_corr
0             0.651851           0.114493

In [11]:
SELECT finance.product_id,
    SUM(finance.revenue / finance.sale_price) AS units_sold,
    AVG(reviews.reviews) AS average_reviews,
    average_reviews / units_sold AS Reviews_per_Unit_Sold
FROM "finance.csv" AS finance
JOIN "reviews.csv" AS reviews
ON   finance.product_id = reviews.product_id
WHERE finance.revenue IS NOT NULL AND finance.revenue != 0
GROUP BY 
    finance.product_id
ORDER BY 
    units_sold DESC;

product_id  units_sold  average_reviews  Reviews_per_Unit_Sold
0     310805-137  401.400000            223.0               0.555556
1     880848-005  189.000000            105.0               0.555556
2         CM0070  178.200257             99.0               0.555555
3         CG3552  178.200200             99.0               0.555555
4         AW4591  178.200105             99.0               0.555555
...          ...         ...              ...                    ...
2903      CJ8087    1.799920              1.0               0.555580
2904      BB9775    1.799917              1.0               0.555581
2905      CL7308    1.799889              1.0               0.555590
2906      CM0051    1.799778              1.0               0.555624
2907      CM0106    1.799599              1.0               0.555679

[2908 rows x 4 columns]

The analysis revealed a moderate correlation of 0.65 between revenue and reviews, indicating that as revenue increases, so does the number of reviews. However, the correlation between revenue and ratings was much weaker, at 0.11. Despite the correlation, it’s important to note that correlation does not imply causation.

Additionally, we observed that for each product, there is roughly one review for every two units sold, which raises concerns about the reliability of the data. The consistency across different products, with similar review-to-sales ratios, suggests that these figures may not be fully reflective of actual customer engagement.

## 5. Top Revenue Sneakers Analysis

In this section, we focused on identifying the sneakers with the highest revenue by analyzing the revenue of each product type, including the number of units sold and their respective ratings. The goal was to determine which sneakers contributed the most to total revenue, while also considering how many units were sold and the relationship with customer ratings. By examining this data, we aimed to uncover insights about product performance and customer preferences based on sales and ratings.

In [13]:
SELECT info.column0,
    SUM(finance.revenue / finance.sale_price) AS units_sold,
    CEIL((SUM(finance.revenue))) AS "Revenue",
	reviews.rating AS Rating
	
FROM "finance.csv" AS finance
JOIN "reviews.csv" AS reviews
ON   finance.product_id = reviews.product_id
JOIN "info.csv" AS info
ON info.column1 = finance.product_id
WHERE finance.revenue IS NOT NULL AND finance.revenue != 0
GROUP BY 
    info.column0, Rating
ORDER BY 
    "Revenue" DESC
LIMIT 10;

column0  ...  Rating
0                                Air Jordan 10 Retro  ...     4.7
1      Unisex Originals CRAIG GREEN KONTUUR II SHOES  ...     2.4
2       Unisex Originals CRAIG GREEN KONTUUR I SHOES  ...     4.1
3  Men's adidas Running Universal Works Ultraboos...  ...     3.9
4                Women's Running Ultraboost 20 Shoes  ...     4.5
5            Men's adidas Originals ZX 4000 4D Shoes  ...     3.1
6                  Men's Running Ultraboost PB Shoes  ...     2.6
7                Women's Running Ultraboost 20 Shoes  ...     2.8
8  UNISEX adidas Originals Pharrell Williams Hu N...  ...     2.7
9  Women's adidas by Stella Mccartney Running Ult...  ...     3.5

[10 rows x 4 columns]

The analysis of the sneakers with the highest revenue revealed interesting findings. Despite Nike not having as many units sold, the Air Jordan 10 Retro took the lead in revenue, with a significant contribution from its high listing price. Notably, the relationship between ratings and revenue was not clear, as some shoes with higher ratings did not show a direct increase in revenue. For instance, the Unisex Originals CRAIG GREEN KONTUUR II SHOES had a lower rating (2.4) but still generated notable revenue.


## 6. Product Description Length and Its Relationship with Price and Units Sold

In this section, the length of the product descriptions was analyzed by grouping them into categories of every 100 characters. The objective was to determine how the description length influences both the number of units sold and the average price within each group. The analysis also considered the average revenue for each group to establish a clearer connection between product description length and its market performance. 

In [27]:
SELECT TRUNC(Length(info.column2)/100.0) *100 as description_length,
		CEIL(SUM(finance.revenue / finance.sale_price)) AS units_sold,
		AVG (finance.revenue) AS average_revenue
FROM "info.csv" AS info
JOIN "finance.csv" AS finance
ON info.column1 = finance.product_id
WHERE info.column2 IS NOT NULL
GROUP BY description_length
ORDER BY description_length;

description_length  units_sold  average_revenue
0                 0.0        44.0       321.480000
1               100.0     29879.0      2459.234632
2               200.0    138579.0      3782.246364
3               300.0     52253.0      4999.201564
4               400.0      9375.0      5346.314672
5               500.0      1534.0     12892.977333
6               600.0      1473.0      9065.792667

In [30]:
SELECT TRUNC(Length(info.column2)/100.0) *100 as description_length,
		AVG (finance.listing_price) AS AVG_listing_price,
		AVG (finance.revenue) AS average_revenue
FROM "info.csv" AS info
JOIN "finance.csv" AS finance
ON info.column1 = finance.product_id
WHERE info.column2 IS NOT NULL
GROUP BY description_length
ORDER BY description_length;

description_length  AVG_listing_price  average_revenue
0                 0.0          33.315000       321.480000
1               100.0          46.784970      2459.234632
2               200.0          68.044268      3782.246364
3               300.0          84.742025      4999.201564
4               400.0         100.480738      5346.314672
5               500.0         125.320667     12892.977333
6               600.0         109.990000      9065.792667

The analysis reveals that products with descriptions between 100 and 300 characters tend to have the highest sales volume. However, the most expensive sneakers, with descriptions ranging from 400 to 600 characters, generate the highest average revenue. While longer descriptions are linked to higher listing prices and revenue, it is the sneakers with descriptions in the 100-300 character range that dominate in units sold. This suggests that, while detailed descriptions may support higher prices, sneakers with shorter descriptions appeal to a broader consumer base, significantly influencing their sales volume.

## Conclusion

The analysis highlights several key insights from the dataset. Adidas dominates in terms of variety, with 52 different price points compared to Nike's 25. Segmenting products by price reveals that most units fall under the "Budget" category, with Adidas also leading in discount distribution. Despite a correlation between revenue and reviews, the lack of a clear cause-and-effect relationship raises doubts. Notably, the Air Jordan 10, leads in revenue. Additionally, while sneakers with descriptions between 100 and 300 characters lead in sales, the highest revenue comes from sneakers with descriptions between 400 and 600 characters. 